genome-alignment-star/alignSTAR.py

```diff
@@ -74,55 +128,74 @@ def main():
             sys.exit('Error: number of input files %s needs to be exactly 1!.' % str(args.input_files))
         generate_fastq_from_ubam(args.input_files[0], outdir, mem=args.mem)
         fqr1 = glob.glob(os.path.join(outdir, '*_1.fastq.gz'))
-        print(fqr1)
         fqr2 = []
+        for fq in fqr1:
+            _fqr2 = fq[:-10] + '2.fastq.gz'
+            if os.path.exists(_fqr2):
+                fqr2.append(_fqr2)
+            else:
+                fqr2.append(None)
+        rgs_input = [os.path.basename(_)[:-11] for _ in fqr1]
+
+        ### we can not have more than one read group per given ubam file
+        assert len(rgs_input) == 1, 'Error: the number of read groups present in given bam file %s is different from 1' % args.input_files[0]
+        
+        ### make sure that the read group present in the ubam also is given in the metadata
+        assert rgs_input[0] in args.readgroup, 'Error: readgroup provided in input bam (%s) is not present in the list of readgroups from metadata (%s)' % (args.input_files[0], str(args.readgroup))
+        rg_idx = args.readgroup.index(rgs_input[0])
+        args.readgroup = args.readgroup[rg_idx]
+        args.pair_status = args.pair_status[rg_idx]
+        fqr1 = fqr1[rg_idx]
```

Comment:
@lindaxiang I have now implemented the following logic for the two use cases:
1. The input is in bam format (that is already pre-processed). Here I assume that either the read group ID provided via `submitter_read_group_id` or the ID provided via `read_group_id_in_bam` is present in this bam. If the latter, I will replace that ID with the one provided in `submitter_read_group_id` for the respective read group. 
2. The input is in fastq format. Here I derive the read group name from the fastq sample name. If the RGID derived from the fastq name is the same as provided via `read_group_id_in_bam`, then I will substitute the RGID with the one provided in `submitter_read_group_id`, otherwise I keep the derived RGID.

Was this the behavior you intended? For the second use case, I find this use of parameters a bit confusing. It would appear more intuitive to me if the `read_group_id_in_bam` is only used for inputs of type BAM and for fastq inputs instead one checks whether there is a read group that has matching `file_r1` and `file_r2` information. If yes, then one can substitute the `submitter_read_group_id` for the one derived from the fastq file name.

Reply:
@akahles You are absolutely right. 
Field: `read_group_id_in_bam` in the payload is only populated when the **whole alignment workflow** inputs are in `BAM` format, if inputs are in `FASTQ` format, this field was empty, and `submitter_read_group_id` has the `RGID` information of FASTQ inputs.

Reply:
Ok, sounds good @lindaxiang . As the metadata can have multiple read groups, I need to do some matching between fastq file names given and at least `file_r1` in the read group metadata, so I can assign the right `submitter_read_group_id`.

Reply:
@lindaxiang This is what I implemented now:
1. For `bam` input, the `read_group_id_in_bam` is also allowed to be present in the input file. In any case, the read group ID from `submitter_read_group_id` will be used for the output.
2. For `fastq` input, the file name of the first mate is checked against `file_r1` to assign the correct read group and then `submitter_read_group_id` from that read group is used for the output. Otherwise, the RGID is derived from the file name.

I made the `read_group_id_in_bam` business metadata-file only, that is it can not be provided via the command line. Let me know if this should be needed, but it would have complicated matters. For your purpose, I assume usage from metadata file is sufficient.

Reply:
This is great!